src/components/MultiStepIndicator/components/StepIndicator/StepIndicator.tsx

```diff
@@ -18,48 +17,79 @@ import Clickable from "../../../../components/Clickable/Clickable";
 import { SIZES, MULTI_STEP_TYPES, STEP_STATUSES } from "../../MultiStepConstants";
 import styles from "./StepIndicator.module.scss";
 import classNames from "classnames";
+import { SubIcon, VibeComponentProps } from "../../../../types";
+import { IconType } from "../../../Icon/IconConstants";
 
 const KEYS = [keyCodes.ENTER, keyCodes.SPACE];
 
-const StepCircleDisplay = ({
+interface StepCircleDisplayProps {
+  status: STEP_STATUSES;
+  isFulfilledStepDisplayNumber: boolean;
+  fulfilledStepIcon: SubIcon;
+  fulfilledStepIconType: IconType.SVG | IconType.ICON_FONT;
+  stepNumber: number;
+  size: SIZES;
+}
+
+const StepCircleDisplay: React.FC<StepCircleDisplayProps> = ({
   status,
   isFulfilledStepDisplayNumber,
   fulfilledStepIcon,
   fulfilledStepIconType,
-  stepNumber,
-  size
+  stepNumber
 }) => {
-  return status === STEP_STATUSES.FULFILLED && !isFulfilledStepDisplayNumber ? (
-    <Icon
-      icon={fulfilledStepIcon}
-      className={classNames(styles.numberContainerTextCheckIcon, { [styles.compact]: size === SIZES.COMPACT })}
-      iconLabel={STEP_STATUSES.FULFILLED}
-      iconType={fulfilledStepIconType}
-      ignoreFocusStyle
-      clickable={false}
-      ariaHidden={true}
-    />
-  ) : (
-    stepNumber
+  return (
+    <>
```

Comment:
Why it was needed?

Reply:
since I wanted `StepCircleDisplay` to always return JSX, not a JSX | number, as it causes TS issues on render

Reply:
check if I can declare 'as' casting for the number without Fragments

Reply:
@hadasfa not possible unfortunately
https://stackoverflow.com/a/70895599/7363120

seems like a TS issue

Though React allows a component to return a number and not necessarily JSX/ReactNode, TS typing does not allow it.. :/